CMakeLists.txt

```diff
@@ -404,6 +405,37 @@ if(LINUX)
   add_definitions(-D__linux__)
 endif(LINUX)
 
+# ASAN and friends
+option(WITH_ASAN "build with ASAN" OFF)
+if(WITH_ASAN)
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fsanitize=address -fno-omit-frame-pointer -lasan")
```

Comment:
I suggest we could add "-fsanitize=leak" which is part of asan. it would be cheaper than varlgrind used currently. maybe we could have a debug mode to enable asan and ubsan to run unittest.